assets/js/util/tracking/index.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * Event tracking utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const DEFAULT_CONFIG = {
+	isFirstAdmin: false,
+	trackingEnabled: false,
+	trackingID: '',
+	referenceSiteURL: '',
+	userIDHash: '',
+};
+
+/**
+ * Tracking configuration.
+ */
+const config = { ...DEFAULT_CONFIG };
+
+/**
+ * The promise created when tracking is enabled.
+ * This reference is kept and returned for subsequent calls.
+ */
+let enableTrackingPromise;
+
+/**
+ * Data layer global used for internal/private Site Kit data.
+ */
+export const DATA_LAYER = '_googlesitekitDataLayer';
+
+/**
+ * Initializes tracking.
+ *
+ * @param {Object} _global The global object. Optional. (Used for testing only)
+ * @param {boolean} _resetPromise Clears the internal reference to the enableTrackingPromise (Optional - for testing only)
+ */
+export async function bootstrapTracking( _global = global, _resetPromise = false ) {
+	const {
+		isFirstAdmin,
+		trackingID,
+		trackingEnabled,
+		referenceSiteURL,
+		userIDHash,
+	} = _global._googlesitekitBase || DEFAULT_CONFIG;
+
+	Object.assign( config, DEFAULT_CONFIG, {
+		isFirstAdmin,
+		trackingID,
+		trackingEnabled,
+		userIDHash,
+		// Remove any trailing slash from the reference URL.
+		referenceSiteURL: referenceSiteURL.toString().replace( /\/+$/, '' ),
```

Comment:
Yeah, it's not probably needed anymore since we updated the reference URL to always be without a trailing slash in WP, but this does pass through the `googlesitekit_inline_base_data` after that so it's technically possible it could have changed. Basically inconsistencies in the (same) URLs would probably throw a wrench in reporting.

Reply:
Would be worth leaving a comment there explaining that, but sounds good 👍

Reply:
Yeah, I think we should continue to ensure that the slash is stripped here.